### PR TITLE
Implemented Continuous Integration and Deployment (CI/CD) Pipeline

### PR DESCRIPTION
**Pull request by aqeell7**
### Description
ISSUE #3
Migrated the existing CI/CD pipeline from GitLab CI to GitHub Actions. It has automated testing, building, and deployment processes, also added workflow badges to the README.md

### Testing (ignore for documentation update)
as the deployment cannot be tested without env variables and secrets, i have thoroughly checked the code syntax and code flow multiple times

### Type of change
- [ ] Bug fix
- [X] New feature
- [ ] Breaking change
- [X] Documentation update

### Checklist:
- [X] I have read and agree to the [Code of Conduct](https://github.com/reclaimprotocol/.github/blob/main/Code-of-Conduct.md).
- [X] I have signed the [Contributor License Agreement (CLA)](https://github.com/reclaimprotocol/.github/blob/main/CLA.md).
- [X] I have considered the [Security Policy](https://github.com/reclaimprotocol/.github/blob/main/SECURITY.md).
- [X] I have self-reviewed and tested my code.
- [X] I have updated documentation as needed.
- [X] I agree to the [project's custom license](https://github.com/reclaimprotocol/.github/blob/main/LICENSE).

### CODE REVIEW

File: .github/workflows/ci-cd.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: 'current'
+
+      - name: Configure Git
```
Comment (AbdulRashidReshamwala) [inline .github/workflows/ci-cd.yml:21]: since we have moved away from gitlab you can drop the step

File: .github/workflows/ci-cd.yml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        run: git config --global url."https://git-push-pull:$GL_TOKEN@gitlab.reclaimprotocol.org".insteadOf "https://gitlab.reclaimprotocol.org"
+        env:
+          GL_TOKEN: ${{ secrets.GL_TOKEN }}
+
+      - name: Install dependencies
```
Comment (AbdulRashidReshamwala) [inline .github/workflows/ci-cd.yml:26]: add another step that installs optional dependencies
```
npm install koffi
```

File: .github/workflows/ci-cd.yml [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+      - name: Set up AWS CLI
+        run: |
+          sudo apt-get update -y
+          sudo apt-get install zip -y
+          sudo apt-get install awscli -y
+
+      - name: Prepare environment
+        run: echo -e "$ENV" > .env.production
+
+      - name: Replace GL_TOKEN in docker-compose
```
Comment (AbdulRashidReshamwala) [inline .github/workflows/ci-cd.yml:49]: this step is no longer required please remove

**Review by AbdulRashidReshamwala — COMMENTED** [review]
LGTM

**Comment by AbdulRashidReshamwala** [comment]
@aqeell7 for testing github action locally you can use https://github.com/nektos/act

File: README.md [hunk 4]
```diff
@@ -5,6 +5,10 @@
     </div>
 </div>
 
+![Test status](https://github.com/ReclaimProtocol/attestor-core/actions/workflows/ci-cd.yml/badge.svg?job=test)
+
+
+![Build status](https://github.com/ReclaimProtocol/attestor-core/actions/workflows/ci-cd.yml/badge.svg?job=build)
```
[the remark this answers is not in the document]
Comment (aqeell7) [reply README.md:11]: yes

File: .github/workflows/ci-cd.yml (hunk 3 above)
Comment (aqeell7) [reply .github/workflows/ci-cd.yml:49]: ok

**Comment by aqeell7** [comment]
@AbdulRashidReshamwala i am encountering some issues with WebSocket connections while testing the ci/cd, when searched for solution its suggesting to make some changes in src/test files.

File: .github/workflows/ci-cd.yml [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+      - name: Prepare environment
+        run: echo -e "$ENV" > .env.production
+
+      - name: Create deployment package
+        run: |
+          export FILE=pkg-$(date +%s%N).zip
+          zip -r $FILE * .* -x '*.git*'
+
+      - name: Configure AWS credentials
```
Comment (AbdulRashidReshamwala) [inline .github/workflows/ci-cd.yml:60]: replace with 
```
          touch ~/.aws/credentials
          printf "[default]\nregion=ap-south-1\naws_access_key_id = %s\naws_secret_access_key = %s\n" "${{ secrets.AWS_ACCESS_KEY_ID }}" "${{ secrets.AWS_SECRET_ACCESS_KEY }}" >> ~/.aws/credentials
          touch ~/.aws/config
```

**Comment by AbdulRashidReshamwala** [comment]
@aqeell7 you can use this as reference 
https://github.com/reclaimprotocol/attestor-core/commit/4789b2d7978f18bddac9bb8596532270064df03d

**Comment by aqeell7** [comment]
@AbdulRashidReshamwala tried using that command but no help, got similar output

**Comment by aqeell7** [comment]
Hello @AbdulRashidReshamwala, if you could review the pull request, I would be grateful.

**Comment by adithyaakrishna** [comment]
@aqeell7 Im taking a look at this, will test and merge it soon

**Pull request merged**